Server/mods/deathmatch/utils/CZipMaker.cpp

```diff
@@ -66,14 +88,23 @@ bool CZipMaker::IsValid()
 // Returns false if failed
 //
 ///////////////////////////////////////////////////////////////
-bool CZipMaker::Close()
+bool CZipMaker::Close() noexcept
 {
-    if (!m_uzFile)
+    if (!IsValid())
         return false;
 
-    int iResult = zipClose(m_uzFile, NULL);
-    m_uzFile = NULL;
-    return iResult == ZIP_OK;
+    zip_close(m_uzFile);
```

Comment:
It is nulled internally

Reply:
> It is nulled internally

How? CZipMaker::m_uzFile does not change in this function.

Reply:
zipClose nulls it internally

Reply:
zip_close does not change the m_uzFile. It is simply impossible because zip_close obtains an argument by value, not by a pointer of a reference. But even if that would be possible it is a bad design allows dependent objects modify internal states of a parent.